src/shaders/symbol_icon.vertex.glsl

```diff
@@ -84,7 +94,15 @@ void main() {
     vec4 projected_pos = u_label_plane_matrix * vec4(a_projected_pos.xy, 0.0, 1.0);
     gl_Position = u_coord_matrix * vec4(projected_pos.xy / projected_pos.w + rotation_matrix * (a_offset / 32.0 * fontScale), 0.0, 1.0);
 
-    v_tex = a_tex / u_texsize;
+    v_tex = a_tex;
+    if (u_crisp && (segment_angle + symbol_rotation == 0.0)) {
+        // move texture loopup to snap to the pixel grid
+        vec4 icon_pos = u_coord_matrix * vec4(projected_pos.xy / projected_pos.w, 0.0, 1.0);
+        v_tex += absfract((icon_pos.xy + 1.0) * u_canvas_size * vec2(1.0, -1.0) / 2.0);
+        v_tex += fract(a_offset / 32.0 * fontScale * u_device_pixel_ratio);
```

Comment:
Is there a need to multiply this by u_coord_matrix before fract, as it is in gl_Position calculation?